### ISSUE TITLE
Feature Request: Histogram Normalisation Parameter

### ISSUE THREAD

**Issue by ZaydH** [piccolomo/plotext#76]
Thanks for creating the great library.

Like `plotext`, `matplotlib` also provides a histogram feature.  One additional parameter `matplotlib` provides is the ability to assign weights to each sample -- particularly useful if you want to plot the histogram as a density plot rather than use the raw counts.  For example, see this [stack overflow discussion](https://stackoverflow.com/questions/3866520/plot-a-histogram-such-that-bar-heights-sum-to-1-probability) of the capability.  

Would this be a feature you think works for `plotext`?

**Comment by ZaydH** [comment]
`plotext`'s interface is lighterweight than `matplotlib` so a simple `density` argument may be a better fit than the full `weights` parameter.  Depending on your thoughts, I may be able to submit a PR as well (assuming you are open to that).

**Comment by ZaydH** [comment]
If we went the route of simply `density`, the API change to [hist](https://github.com/piccolomo/plotext/blob/805c31802c80c9b2b8f4c6231eb8fa603cde7ee4/plotext/_core.py#L250) is quite minimal:
```
def hist(data,
    bins = None,
    yside = None,
    xside = None,
    marker = None,
    color = None,
    fill = True,
    width = None,
    orientation = None,
    label = None, 
    density = False):
    figure.subplot.draw_hist(data,
    bins = bins,
    xside = xside,
    yside = yside,
    marker = marker,
    color = color,
    fill = fill,
    width = width,
    orientation = orientation,
    label = label,
    density = density)
```

From there, [draw_hist](https://github.com/piccolomo/plotext/blob/805c31802c80c9b2b8f4c6231eb8fa603cde7ee4/plotext/_subplot.py#L550) needs a very small change (see the new `if` block):
```
    def draw_hist(self, data, **kwargs):
        bins = kwargs.get("bins")
        bins = self.default.hist_bins if bins is None else bins
        x, y = hist_data(data, bins)
        if kwargs.get("density", False):
            tot = sum(y)
            y = [y_i / tot for y_i in y]
        self.draw_single_bar(x, y, **kwargs)
```

Basic testing of this code on my local setup seems to be good.  I did not look into integrating it into your test suite, but I could.

**Comment by ZaydH** [comment]
`matplotlib` description of the `density` feature:

> The `density` parameter, which normalizes bin heights so that the integral of the histogram is 1. The resulting histogram is an approximation of the probability density function.

[Source](https://matplotlib.org/stable/gallery/statistics/histogram_features.html)

**Comment by piccolomo** [comment]
Hi @ZaydH,

all good on me, you could go ahead or if you prefer I will do it myself (as you want). Here are my suggestions:
 - I would call the parameter `norm` or `normalise`, but really it is up to you. 
 - I would put this parameter right after `data, bins, width` so ye also `width` needs to be changed position (my mistake).
 - this is more relevant: I would add the new parameter to [`hist_data` ](https://github.com/piccolomo/plotext/blob/805c31802c80c9b2b8f4c6231eb8fa603cde7ee4/plotext/_utility/bar.py#L59) and change its code in the [bar.py](https://github.com/piccolomo/plotext/blob/master/plotext/_utility/bar.py#L59) file. 
 - in this case there shouldn't be a need for `if kwargs.get("density", False):` I think,
 - this will require a version update to 4.2.0
 - afterwards we could test it, and i will publish it on PyPi and add you to credits.
 - also I am working on next version (5.0) and i hope i will not mess the repository this time, anyway you will be credited. 

Thanks for your help :-)

**Comment by piccolomo** [comment]
@ZaydH  If this is too boring to do, I could do it

**Comment by ZaydH** [comment]
Sorry for the delay.  I am working on a paper review rebuttal, but I wanted to get this to you before Monday.  

Summary of the Changes:
* Parameter Name: `norm` I would have preferred the verb form, but from your comment, I learned that (American) "normalize" has an alternate British English spelling so to avoid user confusion I went with shorter version with no alternate spellings.
* I still do a missing parameter check (`norm = kwargs.get("norm", False)`) on `norm` in `draw_hist`.  I considered that some (more advanced) users may use the lower-level functions directly, and I did not want this change to break any existing code.
* Normalization is performed inside `hist_data` as you suggested.  I agree that's the cleaner, better way (I was using `hist_data` in my code to calculate the bin counts to get the y-axis ticks.  
* I do not check for divide by zero in norm since you (implicitly) do that check already [at the top of the function](https://github.com/ZaydH/plotext/blob/ce78214349a6ac3654a3575a188da5fd33ec77c6/plotext/_utility/bar.py#L61).
* No issue with `draw_single_bar`.  Your `get_bar_parameters` function just extracts the parameters it needs and ignores the rest.

I tested my revised branch in my (simple) setup and it worked as expected.
![image](https://user-images.githubusercontent.com/6292018/154893636-30295c50-d9ef-453c-bce4-b603ae2cd5d6.png)
